### PR TITLE
Add Documentation to Services Folder files

### PR DESCRIPTION
**Pull request by dchen97**
* Added JSDoc comments according to Google's `JavaScript Style Guide` to all class files in the `client/services` folder (including the `code_evaluators` and `code_preprocessors` subfolders)

### CODE REVIEW

File: client/services/FeedbackGeneratorService.js [hunk 1]
```diff
@@ -92,6 +115,16 @@ tie.factory('FeedbackGeneratorService', [
...
     };
 
+    /**
+     * Returns the Feedback object created as a result of a specified
+     * correctness test.
+     *
+     * @param {string|null} outputFunctionName If needed, is string of the
+     *    name for the function needed to process/format the user's output.
+     * @param {CorrectnessTest} correctnessTest
+     * @param {*} observedOutput Actual output for running user's code.
```
Comment (theresa-lee) [inline client/services/FeedbackGeneratorService.js]: Missing return

File: client/services/FeedbackGeneratorService.js [hunk 2]
```diff
@@ -117,6 +150,12 @@ tie.factory('FeedbackGeneratorService', [
       return feedback;
     };
 
+    /**
+     * Returns the Feedback related to a failing performance test.
+     *
+     * @param {string} expectedPerformance
+     * @private
```
Comment (theresa-lee) [inline client/services/FeedbackGeneratorService.js]: Missing return

File: client/services/SolutionHandlerService.js [hunk 3]
```diff
@@ -26,7 +26,19 @@ tie.factory('SolutionHandlerService', [
...
-      // Returns a promise with a Feedback object.
+      /**
+       * Asynchronously returns a Promise with a Feedback object associated
+       * with the code submission and the test results.
+       *
+       * @param {Array} tasks
+       * @param {string} starterCode Code given at the beginning before user
+       *    adds their own code.
+       * @param {string} studentCode Code that the user submitted
+       * @param {string} auxiliaryCode Code thqt the user does not write but
```
Comment (theresa-lee) [inline client/services/SolutionHandlerService.js]: that*

File: client/services/SolutionHandlerService.js [hunk 4]
```diff
@@ -26,7 +26,19 @@ tie.factory('SolutionHandlerService', [
...
+       *    is needed to for the code to pass the given tests.
```
Comment (theresa-lee) [inline client/services/SolutionHandlerService.js]: is needed to pass the given tests ? (have some extra words in there)

File: client/services/code_evaluators/PythonPrereqCheckService.js [hunk 5]
```diff
@@ -25,13 +25,23 @@ tie.factory('PythonPrereqCheckService', [
       PrereqCheckFailureObjectFactory, PREREQ_CHECK_TYPE_BAD_IMPORT,
       PREREQ_CHECK_TYPE_MISSING_STARTER_CODE, SUPPORTED_PYTHON_LIBS,
       PREREQ_CHECK_TYPE_GLOBAL_CODE) {
-
+    /**
+     * Removes trailing white space that may be at the end of a string.
+     *
+     * @param {string} str
```
Comment (theresa-lee) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: missing return

File: client/services/code_evaluators/PythonPrereqCheckService.js [hunk 6]
```diff
@@ -56,6 +73,13 @@ tie.factory('PythonPrereqCheckService', [
       return topLevelFunctionLines;
     };
 
+    /**
+     * Checks if the code submission has any function signatures that are
+     * different or missing from the expected function signatures.
+     *
+     * @param {string} code
+     * @param {string} expectedTopLevelFunctionLines
```
Comment (theresa-lee) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: missing return?

File: client/services/code_evaluators/PythonPrereqCheckService.js [hunk 7]
```diff
@@ -68,12 +92,26 @@ tie.factory('PythonPrereqCheckService', [
       });
     };
 
+    /**
+     * Checks if the code given at the beginning - before the user began making
+     * changes - is still present and unaltered.
+     *
+     * @param starterCode
+     * @param code
```
Comment (theresa-lee) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: The type missing and missing return?

File: client/services/code_evaluators/PythonPrereqCheckService.js [hunk 8]
```diff
@@ -87,14 +125,29 @@ tie.factory('PythonPrereqCheckService', [
       return importedLibraries;
     };
 
+    /**
+     * Returns an array of strings each containing the name of a library the
+     * user tried to import in their submission but is not supported in the TIE
+     * system.
+     *
+     * @param {Array} importedLibraries
```
Comment (theresa-lee) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: missing return

File: client/services/FeedbackGeneratorService.js (hunk 1 above)
Comment (dchen97) [inline client/services/FeedbackGeneratorService.js]: Done!

File: client/services/FeedbackGeneratorService.js (hunk 2 above)
Comment (dchen97) [inline client/services/FeedbackGeneratorService.js]: Done!

File: client/services/SolutionHandlerService.js (hunk 3 above)
Comment (dchen97) [inline client/services/SolutionHandlerService.js]: Done!

File: client/services/SolutionHandlerService.js (hunk 4 above)
Comment (dchen97) [inline client/services/SolutionHandlerService.js]: Done!

File: client/services/code_evaluators/PythonPrereqCheckService.js (hunk 5 above)
Comment (dchen97) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: Done!

File: client/services/code_evaluators/PythonPrereqCheckService.js (hunk 6 above)
Comment (dchen97) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: Done!

File: client/services/code_evaluators/PythonPrereqCheckService.js (hunk 8 above)
Comment (dchen97) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: Done!

File: client/services/code_evaluators/PythonPrereqCheckService.js (hunk 7 above)
Comment (dchen97) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: Done!

**Pull request merged**